verilog/analysis/verilog_project.cc

```diff
@@ -345,4 +346,51 @@ absl::StatusOr<FileList> ParseSourceFileListFromFile(
   return ParseSourceFileList(file_list_file, content);
 }
 
+absl::StatusOr<FileList> ParseSourceFileListFromCommandline(
+    const std::vector<absl::string_view>& cmdline) {
+  FileList file_list_out;
+  for (absl::string_view argument :
+       verible::make_range(cmdline.begin() + 1, cmdline.end())) {
+    // cmdline[0] is the tool's name, which can be skipped.
+    if (argument.empty()) continue;
+    if (argument[0] != '+') {
+      // Then "argument" is a SV file name.
+      file_list_out.file_paths.push_back(std::string(argument));
+    } else {
+      // it should be either a define or incdir.
+      std::vector<absl::string_view> argument_plus_splitted =
+          absl::StrSplit(argument, absl::ByChar('+'), absl::SkipEmpty());
+      if (argument_plus_splitted.size() < 2) {
+        return absl::InvalidArgumentError("Unkown argument.");
```

Comment:
Maybe we can make the message so that the user, who just typed a wrong command line, has some actionable information

```c++
  return absl::InvalidArgumentError(absl::StrCat("expected either '+define+' or '+incdir+', but got '", argument, "'"));
```

Reply:
Can you double-check how we treat thesse situation in the `FileList` parser from a file ? A defensive strategy might as well be to just treat `argument` as a `file_path` (in case someone has a crazy filename `+foobar/baz.sv`), and then just have further stages that attempt to open the file complain about its non-existence.

Anyway, just make sure that we are consistent with what would happen in the file list parser parsing and command line parsing.

Reply:
They are both on the same page, as both don't check if the SystemVerilog file or `incdir` exists, it just adds it.
I would only add defines parsing in `ParseSourceFileList()`.

Reply:
Also judging by the way it looks `ParseSourceFileList()` doesn't support `+incdir+dir1+dir2`, it only removes the prefix.

Reply:
alright, so some TODO to improve the ParseSourceFileList() at some point.